File: generate-icons.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const sizes = [72, 96, 128, 144, 152, 192, 384, 512];
const inputFile = path.join(__dirname, 'public', 'fbo-logo.jpg');
const outputDir = path.join(__dirname, 'public');

async function generateIcons() {
  console.log('Generating PWA icons from fbo-logo.jpg...');
  
  for (const size of sizes) {
    const outputFile = path.join(outputDir, `icon-${size}x${size}.png`);
    
    try {
      await sharp(inputFile)
        .resize(size, size, {
          fit: 'cover',
          position: 'center'
        })
        .png()
        .toFile(outputFile);
      
      console.log(`✓ Generated ${size}x${size} icon`);
    } catch (error) {
      console.error(`✗ Failed to generate ${size}x${size}:`, error.message);
    }
  }
  
  // Generate apple-touch-icon
  try {
    await sharp(inputFile)
      .resize(180, 180, {
        fit: 'cover',
        position: 'center'
      })
      .png()
      .toFile(path.join(outputDir, 'apple-touch-icon.png'));
    
    console.log('✓ Generated apple-touch-icon.png');
  } catch (error) {
    console.error('✗ Failed to generate apple-touch-icon:', error.message);
  }
  
  // Generate favicon
  try {
    await sharp(inputFile)
      .resize(32, 32, {
        fit: 'cover',
        position: 'center'
      })
      .png()
      .toFile(path.join(outputDir, 'favicon.png'));
    
    console.log('✓ Generated favicon.png');
  } catch (error) {
    console.error('✗ Failed to generate favicon:', error.message);
  }
  
  console.log('\n✅ All icons generated successfully!');
}

generateIcons().catch(console.error);

File: generate_icons.py
#!/usr/bin/env python3
from PIL import Image
import os

# Paths
input_file = '/home/owenoz123/Desktop/kawogo-web/public/fbo-logo.jpg'
output_dir = '/home/owenoz123/Desktop/kawogo-web/public'

# Icon sizes
sizes = [72, 96, 128, 144, 152, 192, 384, 512]

print('Generating PWA icons from fbo-logo.jpg...')

# Open the source image
img = Image.open(input_file)

# Generate icons
for size in sizes:
    output_file = os.path.join(output_dir, f'icon-{size}x{size}.png')
    resized = img.resize((size, size), Image.Resampling.LANCZOS)
    resized.save(output_file, 'PNG')
    print(f'✓ Generated {size}x{size} icon')

# Generate apple-touch-icon (180x180)
apple_icon = img.resize((180, 180), Image.Resampling.LANCZOS)
apple_icon.save(os.path.join(output_dir, 'apple-touch-icon.png'), 'PNG')
print('✓ Generated apple-touch-icon.png')

# Generate favicon (32x32)
favicon = img.resize((32, 32), Image.Resampling.LANCZOS)
favicon.save(os.path.join(output_dir, 'favicon.png'), 'PNG')
print('✓ Generated favicon.png')

print('\n✅ All icons generated successfully!')
